pilot/pkg/model/xds_cache.go

```diff
@@ -164,20 +188,67 @@ func newLru(evictCallback simplelru.EvictCallback) simplelru.LRUCache {
 	return l
 }
 
-func (l *lruCache) evict(k interface{}, v interface{}) {
+// This is the callback passed to LRU, it will be called whenever a key is removed.
+func (l *LruCache) onEvict(k interface{}, _ interface{}) {
 	if features.EnableXDSCacheMetrics {
 		xdsCacheEvictions.Increment()
 	}
 
-	key := k.(string)
-	value := v.(cacheValue)
+	if l.trackEvicted {
```

Comment:
in what scenario do we not want to track evicted? do we just not evict if this is not set?

Reply:
Actively `Remove`